### PR TITLE
Product type fix

### PR DESCRIPTION
**Pull request by igormukhingmailcom**

### CODE REVIEW

**Review by loevgaard — APPROVED** [review]

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -30,6 +30,15 @@ public function getConfigTreeBuilder(): TreeBuilder
...
                 ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->end()
+                ->arrayNode('generation')
+                    ->addDefaultsIfNotSet()
+                    ->children()
+                        ->booleanNode('exclude_root_taxon')
+                            ->info('Exclude root taxon while generating product type?')
+                            ->defaultFalse()
+                        ->end()
+                    ->end()
+                ->end()
```
Comment (loevgaard) [inline src/DependencyInjection/Configuration.php:41]: Remove this again. This is very related to the specific feed type, i.e. the Google feed. In the future we should have such configuration when creating the feed

File: src/DependencyInjection/SetonoSyliusFeedExtension.php [hunk 2]
```diff
@@ -21,6 +21,7 @@ public function load(array $config, ContainerBuilder $container): void
         $config = $this->processConfiguration($this->getConfiguration([], $container), $config);
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
 
+        $container->setParameter('setono_sylius_feed.generation.exclude_root_taxon', $config['generation']['exclude_root_taxon']);
```
Comment (loevgaard) [inline src/DependencyInjection/SetonoSyliusFeedExtension.php:24]: Remove this

File: src/FeedContext/Google/Shopping/ProductItemContext.php [hunk 3]
```diff
@@ -60,8 +68,13 @@ public function getContextList(object $product, ChannelInterface $channel, Local
             ));
         }
 
-        $productType = $this->getProductType($product);
+        if ($product instanceof ProductTypeAwareInterface) {
```
Comment (loevgaard) [inline src/FeedContext/Google/Shopping/ProductItemContext.php:71]: The name 'Product type' comes from Google's lingo. If we should have an interface like this, it should be named something else: 'Breadcrumb', 'Taxon Path' or something. I probably like taxon path the most

**Review by loevgaard — CHANGES_REQUESTED** [review]

File: src/DependencyInjection/Configuration.php (hunk 1 above)
Comment (igormukhingmailcom) [reply src/DependencyInjection/Configuration.php:41]: Ok

File: src/FeedContext/Google/Shopping/ProductItemContext.php (hunk 3 above)
Comment (igormukhingmailcom) [reply src/FeedContext/Google/Shopping/ProductItemContext.php:71]: Ok

**Review by loevgaard — APPROVED** [review]

**Pull request merged**